src/main/kotlin/org/bilalkilic/kediatrhelper/listeners/KediatrHandlerClassMarker.kt

```diff
@@ -51,9 +63,41 @@ class KediatrHandlerClassMarker : LineMarkerProvider {
     }
 
     private fun navigationHandler(superClassQualifiedNames: Collection<String>, commandTypeName: String) =
-        GutterIconNavigationHandler<PsiElement> { _, elt ->
+        GutterIconNavigationHandler<PsiElement> { me, elt ->
             val project = elt.project
             val handler = project.service<HandlerService>().findHandler(elt, superClassQualifiedNames, commandTypeName)
-            handler.firstOrNull()?.navigate(false)
+            handler.firstOrNull()?.navigate(false) ?: showPopupWithIcon(elt, me)
         }
+
+    private fun showPopup(element: PsiElement, me: MouseEvent) {
+        val mainClass = (element.parent as KtFile).classes.firstOrNull() ?: return
+        val title = POPUP_TITLE_PREFIX + mainClass.name
+        val popup = JBPopupFactory.getInstance()
+            .createPopupChooserBuilder<Any>(listOf(" + $POPUP_STEP_MESSAGE"))
+            .setTitle(title)
+            .setMovable(false)
+            .setResizable(false)
+            .setRequestFocus(true)
+            .setCancelOnWindowDeactivation(false)
+            .setItemChosenCallback { element.project.service<CreateNewHandlerService>().create(element) }
+            .createPopup()
+        popup.show(RelativePoint(Point(me.xOnScreen + 30, me.yOnScreen + 20)))
+    }
+
+    private fun showPopupWithIcon(element: PsiElement, me: MouseEvent) {
+        val mainClass = (element.parent as KtFile).classes.firstOrNull() ?: return
+        val title = POPUP_TITLE_PREFIX + mainClass.name
+        val options = listOf(POPUP_STEP_MESSAGE)
+        val step = object : BaseListPopupStep<String>(title, options, Icons.createNewHandlerGutter) {
+            override fun onChosen(selectedValue: String, finalChoice: Boolean): PopupStep<*>? {
```

Comment:
Maybe we could add creating async handlers as an option too.

Reply:
added.